Address safer cpp warnings in PageClientImplCocoa.mm

cdumez opened the pull request:
#### 83f31203852abe9f250efec1f54e2dfdb6e11342
<pre>
Address safer cpp warnings in PageClientImplCocoa.mm
<a href="https://bugs.webkit.org/show_bug.cgi?id=300072">https://bugs.webkit.org/show_bug.cgi?id=300072</a>

Reviewed by Ryosuke Niwa.

* Source/WebKit/SaferCPPExpectations/RetainPtrCtorAdoptCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm:
(WebKit::PageClientImplCocoa::obscuredContentInsetsDidChange):
(WebKit::PageClientImplCocoa::themeColorWillChange):
(WebKit::PageClientImplCocoa::themeColorDidChange):
(WebKit::PageClientImplCocoa::underPageBackgroundColorWillChange):
(WebKit::PageClientImplCocoa::sampledPageTopColorWillChange):
(WebKit::PageClientImplCocoa::sampledPageTopColorDidChange):
(WebKit::PageClientImplCocoa::spatialBackdropSourceWillChange):
(WebKit::PageClientImplCocoa::spatialBackdropSourceDidChange):
(WebKit::PageClientImplCocoa::isPlayingAudioWillChange):
(WebKit::PageClientImplCocoa::isPlayingAudioDidChange):
(WebKit::PageClientImplCocoa::scrollingUpdatesDisabledForTesting):
(WebKit::PageClientImplCocoa::didInsertAttachment):
(WebKit::PageClientImplCocoa::didRemoveAttachment):
(WebKit::PageClientImplCocoa::didInvalidateDataForAttachment):
(WebKit::PageClientImplCocoa::allocFileWrapperInstance const):
(WebKit::PageClientImplCocoa::serializableFileWrapperClasses const):
(WebKit::PageClientImplCocoa::storeAppHighlight):
(WebKit::PageClientImplCocoa::gpuProcessDidFinishLaunching):
(WebKit::PageClientImplCocoa::gpuProcessDidExit):
(WebKit::PageClientImplCocoa::modelProcessDidFinishLaunching):
(WebKit::PageClientImplCocoa::modelProcessDidExit):
(WebKit::PageClientImplCocoa::dictationAlternatives):
(WebKit::PageClientImplCocoa::microphoneCaptureWillChange):
(WebKit::PageClientImplCocoa::cameraCaptureWillChange):
(WebKit::PageClientImplCocoa::displayCaptureWillChange):
(WebKit::PageClientImplCocoa::displayCaptureSurfacesWillChange):
(WebKit::PageClientImplCocoa::systemAudioCaptureWillChange):
(WebKit::PageClientImplCocoa::microphoneCaptureChanged):
(WebKit::PageClientImplCocoa::cameraCaptureChanged):
(WebKit::PageClientImplCocoa::displayCaptureChanged):
(WebKit::PageClientImplCocoa::displayCaptureSurfacesChanged):
(WebKit::PageClientImplCocoa::systemAudioCaptureChanged):
(WebKit::PageClientImplCocoa::windowKind):
(WebKit::PageClientImplCocoa::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebKit::PageClientImplCocoa::proofreadingSessionUpdateStateForSuggestionWithID):
(WebKit::writingToolsActiveKeySingleton):
(WebKit::PageClientImplCocoa::writingToolsActiveWillChange):
(WebKit::PageClientImplCocoa::writingToolsActiveDidChange):
(WebKit::PageClientImplCocoa::didEndPartialIntelligenceTextAnimation):
(WebKit::PageClientImplCocoa::writingToolsTextReplacementsFinished):
(WebKit::PageClientImplCocoa::addTextAnimationForAnimationID):
(WebKit::PageClientImplCocoa::removeTextAnimationForAnimationID):
(WebKit::PageClientImplCocoa::setGamepadsRecentlyAccessed):
(WebKit::PageClientImplCocoa::hasActiveNowPlayingSessionChanged):
(WebKit::PageClientImplCocoa::videoControlsManagerDidChange):
(WebKit::PageClientImplCocoa::platformWindow const):
(WebKit::PageClientImplCocoa::processDidUpdateThrottleState):

Canonical link: <a href="https://commits.webkit.org/300926@main">https://commits.webkit.org/300926@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c2bdd3c0561441f55a4af0115f6d2dadcc9f1163

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124318 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44000 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/34728 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131156 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/76375 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 ios-apple 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44745 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52599 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94567 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/76375 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127272 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35661 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111198 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75154 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/34601 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/74634 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105405 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/29581 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/133823 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51221 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39059 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103046 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/51617 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107417 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102842 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26174 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48202 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26457 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48158 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51086 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/56873 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50522 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/53878 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52197 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm:
```diff
@@ -58,22 +58,23 @@
 void PageClientImplCocoa::themeColorWillChange()
 {
-    [m_webView willChangeValueForKey:@"themeColor"];
+    [webView() willChangeValueForKey:@"themeColor"];
```
Maybe we should rename this to `protectedWebView` in a follow up?

cdumez replied:
We can but it is similar as in my other PR, there would only be a `protectedWebView()` getter, no `webView()` getter then, since `WeakObjCPtr<T>::get()` returns a `RetainPtr`.

rniwa replied:
Hm... I see. I guess because objective-c weakptr is thread safe?

cdumez replied:
I assume you're likely right.

cdumez committed:
Address safer cpp warnings in PageClientImplCocoa.mm
https://bugs.webkit.org/show_bug.cgi?id=300072

Reviewed by Ryosuke Niwa.

* Source/WebKit/SaferCPPExpectations/RetainPtrCtorAdoptCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm:
(WebKit::PageClientImplCocoa::obscuredContentInsetsDidChange):
(WebKit::PageClientImplCocoa::themeColorWillChange):
(WebKit::PageClientImplCocoa::themeColorDidChange):
(WebKit::PageClientImplCocoa::underPageBackgroundColorWillChange):
(WebKit::PageClientImplCocoa::sampledPageTopColorWillChange):
(WebKit::PageClientImplCocoa::sampledPageTopColorDidChange):
(WebKit::PageClientImplCocoa::spatialBackdropSourceWillChange):
(WebKit::PageClientImplCocoa::spatialBackdropSourceDidChange):
(WebKit::PageClientImplCocoa::isPlayingAudioWillChange):
(WebKit::PageClientImplCocoa::isPlayingAudioDidChange):
(WebKit::PageClientImplCocoa::scrollingUpdatesDisabledForTesting):
(WebKit::PageClientImplCocoa::didInsertAttachment):
(WebKit::PageClientImplCocoa::didRemoveAttachment):
(WebKit::PageClientImplCocoa::didInvalidateDataForAttachment):
(WebKit::PageClientImplCocoa::allocFileWrapperInstance const):
(WebKit::PageClientImplCocoa::serializableFileWrapperClasses const):
(WebKit::PageClientImplCocoa::storeAppHighlight):
(WebKit::PageClientImplCocoa::gpuProcessDidFinishLaunching):
(WebKit::PageClientImplCocoa::gpuProcessDidExit):
(WebKit::PageClientImplCocoa::modelProcessDidFinishLaunching):
(WebKit::PageClientImplCocoa::modelProcessDidExit):
(WebKit::PageClientImplCocoa::dictationAlternatives):
(WebKit::PageClientImplCocoa::microphoneCaptureWillChange):
(WebKit::PageClientImplCocoa::cameraCaptureWillChange):
(WebKit::PageClientImplCocoa::displayCaptureWillChange):
(WebKit::PageClientImplCocoa::displayCaptureSurfacesWillChange):
(WebKit::PageClientImplCocoa::systemAudioCaptureWillChange):
(WebKit::PageClientImplCocoa::microphoneCaptureChanged):
(WebKit::PageClientImplCocoa::cameraCaptureChanged):
(WebKit::PageClientImplCocoa::displayCaptureChanged):
(WebKit::PageClientImplCocoa::displayCaptureSurfacesChanged):
(WebKit::PageClientImplCocoa::systemAudioCaptureChanged):
(WebKit::PageClientImplCocoa::windowKind):
(WebKit::PageClientImplCocoa::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebKit::PageClientImplCocoa::proofreadingSessionUpdateStateForSuggestionWithID):
(WebKit::writingToolsActiveKeySingleton):
(WebKit::PageClientImplCocoa::writingToolsActiveWillChange):
(WebKit::PageClientImplCocoa::writingToolsActiveDidChange):
(WebKit::PageClientImplCocoa::didEndPartialIntelligenceTextAnimation):
(WebKit::PageClientImplCocoa::writingToolsTextReplacementsFinished):
(WebKit::PageClientImplCocoa::addTextAnimationForAnimationID):
(WebKit::PageClientImplCocoa::removeTextAnimationForAnimationID):
(WebKit::PageClientImplCocoa::setGamepadsRecentlyAccessed):
(WebKit::PageClientImplCocoa::hasActiveNowPlayingSessionChanged):
(WebKit::PageClientImplCocoa::videoControlsManagerDidChange):
(WebKit::PageClientImplCocoa::platformWindow const):
(WebKit::PageClientImplCocoa::processDidUpdateThrottleState):

Canonical link: https://commits.webkit.org/300926@main
